### PR TITLE
Rev-007-create-login-page

### PR DESCRIPTION
**Pull request by IbrahimAldereni-socuimtech**
LOGIN-PAGE

- figma link:
https://www.figma.com/file/22d3bKcFL0ipWIZJOUY3Km/reverifi?node-id=0%3A15050

- REV-006 signup page PR (base)
https://github.com/abukhadraop-or/reverifi/pull/9

### CODE REVIEW

File: src/pages/Login/login.styles.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  align-items: center;
+  background-color: ${colors.lightgrey2};
+  display: none;
+  flex: 1;
+  justify-content: center;
+
+  ${mq.desktopWide`
+    display: flex;
+  `}
+`;
```
Comment (NoorAbuzainahOP) [inline src/pages/Login/login.styles.js:23]: I noticed that some of the styled components here are duplicate with the ones in `src/pages/SignUp/sign-up.styles.js`, so let's reuse the same components if possible.
I'd suggest moving some components to the shared components directory and make accessible, e.g. we can move this to `src/components/Image`. Same thing goes for the rest of the components

**Comment by NoorAbuzainahOP** [comment]
This probably came up from a previous PR, but some files naming seem kind of redundant e.g. `src/pages/Login/Login.jsx`
would make more sense to use naming convention similar to the below?
```
├── Login
│   ├── index.jsx
│   ├── schema.js
│   ├── style.js
│   └── test.jsx
```

File: src/pages/Login/login.styles.js (hunk 1 above)
Comment (IbrahimAldereni-socuimtech) [reply src/pages/Login/login.styles.js:23]: some styling differs from the signup, that why I separated them

File: src/pages/Login/Login.jsx [hunk 2]
```diff
@@ -1,15 +1,105 @@
...
+
+        <FormContainer onSubmit={handleSubmit(onSubmit)}>
+          <InputLabel htmlFor="email">E-mail</InputLabel>
+          <InputField
+            type="email"
+            {...register('email')}
+            id="email"
+            placeholder="eg: Jhon@domain.com"
+          />
+          <ErrorMessage>{errors.email?.message}</ErrorMessage>
```
Comment (NoorAbuzainahOP) [inline src/pages/Login/Login.jsx:64]: I noticed that every InputField component needs an ErrorMessage, let's create a shred jsx component for both.

File: src/pages/Login/Login.jsx [hunk 3]
```diff
@@ -1,15 +1,105 @@
...
+            {...register('password')}
+            id="password"
+            placeholder="password"
+          />
+          <ErrorMessage>{errors.password?.message}</ErrorMessage>
+
+          <CheckboxInputContainer>
+            <CheckboxInputField type="checkbox" {...register('rememberMe')} />
+            <InputLabel>Remember me</InputLabel>
+          </CheckboxInputContainer>
```
Comment (NoorAbuzainahOP) [inline src/pages/Login/Login.jsx:81]: same here

File: src/pages/Login/Login.jsx [hunk 4]
```diff
@@ -1,15 +1,105 @@
...
+            Continue with Facebook
+          </SocialButton>
+          <SocialButton dark>
+            <AppleIcon />
+            Continue with Apple
+          </SocialButton>
+          <SocialButton light>
+            <GoogleIcon />
+            Continue with Google
+          </SocialButton>
```
Comment (NoorAbuzainahOP) [inline src/pages/Login/Login.jsx:99]: we can use a list of buttons here as well

File: src/pages/Login/Login.jsx [hunk 5]
```diff
@@ -1,15 +1,112 @@
...
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: yupResolver(loginSchema),
+  });
+
+  const onSubmit = () => {
```
Comment (abukhadraop) [inline src/pages/Login/Login.jsx:43]: All functions should have docs

File: src/pages/Login/Login.test.jsx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+describe('test2', () => {
```
Comment (abukhadraop) [inline src/pages/Login/Login.test.jsx:1]: lets remove anything related to unit tests for now

File: package-lock.json [hunk 7]
```diff
@@ -10,6 +10,11 @@
       "dependencies": {
```
Comment (abukhadraop) [inline package-lock.json:10]: please revert all changes on this file

File: src/pages/Login/LoginSchema.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import * as yup from 'yup';
```
Comment (abukhadraop) [inline src/pages/Login/LoginSchema.js:1]: we use kebab case for `.js` files

File: src/pages/Login/LoginSchema.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+const loginSchema = yup
+  .object({
+    email: yup.string().email().required(),
+    password: yup
```
Comment (abukhadraop) [inline src/pages/Login/LoginSchema.js:6]: please add label to these fields as we did in the sign up schema

File: src/pages/SignUp/SignUp.test.jsx [hunk 10]
```diff
@@ -0,0 +1,6 @@
+describe('test', () => {
```
Comment (abukhadraop) [inline src/pages/SignUp/SignUp.test.jsx:1]: please remove this file

File: src/setupTests.js [hunk 11]
```diff
@@ -2,6 +2,8 @@ import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
 import Enzyme from 'enzyme';
```
Comment (abukhadraop) [inline src/setupTests.js:2]: please remove this file

File: src/styles/colors.js [hunk 12]
```diff
@@ -1,11 +1,15 @@
 const colors = {
   alabaster: '#fafafa',
   blue: '#006aff',
+  darkshadow: '#22222229',
```
Comment (abukhadraop) [inline src/styles/colors.js:4]: color opacity does not change the color itself, this file should only contain unique colors to make them reusable, you can append the opacity to the color when you use it, ex:
https://github.com/abukhadraop-or/reverifi/blob/d92893233670c6ecbe5dff52e208d086c02c65ba/src/components/FeaturedListing/Card/card.styles.js#L86

File: src/styles/colors.js [hunk 13]
```diff
@@ -1,11 +1,15 @@
...
   dustyGrey: '#979797',
   gray: '#6f7076',
   green: '#b2d235',
+  grey: '#2222227f',
+  grey2: '#545454',
+  lightblue: '#0084fc',
```
Comment (abukhadraop) [inline src/styles/colors.js:10]: i dont see these colors used anywhere, why they added?

File: src/styles/colors.js [hunk 14]
```diff
@@ -1,11 +1,15 @@
...
   mercury: '#e4e4e4',
-  midGrey: '#d8d8d8',
+  midgrey: '#d8d8d8',
```
Comment (abukhadraop) [inline src/styles/colors.js:12]: we use camelcase naming for object properties, why this renamed to `midgrey` ?

File: src/routes.js [hunk 15]
```diff
@@ -26,7 +26,7 @@ const routes = [
   {
     component: lazy(() => import('pages/SignUp/SignUp')),
     label: 'sign-up',
-    path: '/sign-up',
+    path: '/signUp',
```
Comment (abukhadraop) [inline src/routes.js:29]: please revert this changes

File: src/pages/SignUp/sign-up.styles.js [hunk 16]
```diff
@@ -4,15 +4,13 @@ import mq from 'styles/media-query';
...
   display: flex;
-
   > div {
     flex: 1;
   }
 `;
 
 export const ImageContainer = styled.div`
   display: none;
-
```
Comment (abukhadraop) [inline src/pages/SignUp/sign-up.styles.js:15]: any reason why these empty lines were removed ? i added them to make the styles more readable

File: src/pages/SignUp/sign-up.styles.js [hunk 17]
```diff
@@ -24,7 +22,6 @@ export const ImageContainer = styled.div`
 export const InfoContainer = styled.div`
   flex: 1;
   padding: 1rem;
-
   ${mq.desktopWide`
```
Comment (abukhadraop) [inline src/pages/SignUp/sign-up.styles.js:25]: lets use the mobile breaking point here and in the login page

**Review by abukhadraop — CHANGES_REQUESTED** [review]

**Review by abukhadraop — APPROVED** [review]

**Pull request merged**